vault/expiration.go

```diff
@@ -1269,6 +1274,17 @@ func (m *ExpirationManager) emitMetrics() {
 	num := len(m.pending)
 	m.pendingLock.RUnlock()
 	metrics.SetGauge([]string{"expire", "num_leases"}, float32(num))
+	// Check if lease count is greater than the threshold
+	if num > maxLeaseThreshold {
+		if atomic.LoadUint32(&m.leaseCheckCounter) > 59 {
+			m.logger.Warn("expiration: lease count exceeds maximum lease threshold")
```

Comment:
I think "...exceeds warning lease threshold" is nicer or people might think that they have hit a hard max and might hit issues immediately.